torchao/prototype/quantized_training/int8.py

```diff
@@ -0,0 +1,247 @@
+from typing import Any, Optional, Tuple
+
+import torch
+from torch import Tensor, nn
+from torch.utils._python_dispatch import return_and_correct_aliasing
+
+from torchao.dtypes.utils import _dispatch__torch_dispatch__, _dispatch__torch_function__, _implements
+
+
+aten = torch.ops.aten
+c10d_functional = torch.ops.c10d_functional
+_c10d_functional = torch.ops._c10d_functional
+
+
+# the main difference of this tensor subclass from AffineQuantizedTensor:
+# 1. F.linear is differentiable i.e. backward is defined.
+# 2. support stochastic rounding when casting from floating point.
+class Int8QTLinearWeight(Tensor):
```

Comment:
Can you add a docstring to include some details on what kind of int8 quant this is doing (e.g. symmetric, -127 to 127)? Also you can move the comment above into the docstring